smarts/core/sumo_road_network.py

```diff
@@ -969,14 +974,17 @@ def _resolve_in_junction(self, pose: Pose) -> List[str]:
         lane = closest_lps[0].lane
         if not lane.in_junction:
             return []
-        road_ids = [lane.road.road_id]
-        next_roads = lane.road.outgoing_roads
-        assert (
-            len(next_roads) <= 1
-        ), "A junction is expected to have <= 1 outgoing roads"
-        if next_roads:
-            road_ids.append(next_roads[0].road_id)
-        return road_ids
+
+        wp_lanes = []
+        for incoming_lane in lane.incoming_lanes:
+            for junction_lane in incoming_lane.outgoing_lanes:
```

Comment:
This may or may not be enough, I remember that sometimes the junction lanes are subdivided so that the incoming lane is still a junction lane. You may want to do a `in_junction` check to get back to an incoming lane that is not in the junction.

Reply:
Good to know. Should we hold off on merging this until we test it a bit more? I think this change will only help.